File: delivery/src/main/java/br/org/restaurantepopular/delivery/config/AppModule.java
package br.org.restaurantepopular.delivery.config;

import br.org.restaurantepopular.usecase.core.IUseCaseExecutor;
import br.org.restaurantepopular.usecase.core.UseCaseExecutor;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

@Configuration
public class AppModule {

    @Bean
    public IUseCaseExecutor useCaseExecutor() {
        return new UseCaseExecutor();
    }
}

File: delivery/src/main/java/br/org/restaurantepopular/delivery/config/AppModule.kt
package br.org.restaurantepopular.delivery.config

import br.org.restaurantepopular.usecase.core.UseCaseExecutor
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration

@Configuration
class AppModule {
    @Bean
    fun useCaseExecutor() = UseCaseExecutor()
}